x-pack/docs/en/rest-api/security/bulk-update-api-keys.asciidoc

```diff
@@ -2,4 +2,367 @@
+----
+POST /_security/api_key/_bulk_update
+{
+  "ids": [
+    "VuaCfGcBCdbkQm-e5aOx",
+    "H3_AhoIBA9hmeQJdg7ij"
+  ],
+  "role_descriptors": {}
+}
+----
+// TEST[skip:api key ids not available]
+
+Which returns the response:
+
+[source,console-result]
+----
+{
+  "updated": [
+    "VuaCfGcBCdbkQm-e5aOx",
+    "H3_AhoIBA9hmeQJdg7ij"
+  ],
+  "noops": []
+}
+----
+
+The API keys' effective permissions after the update will the same as the <<security-api-bulk-update-api-keys-examples-user-permissions, owner user's>>:
+
+[source,js]
+--------------------------------------------------
+{
+  "cluster": ["all"],
+  "index": [
+    {
+      "names": ["*"],
+      "privileges": ["all"]
+    }
+  ]
+}
+--------------------------------------------------
+// NOTCONSOLE
+
+For the next example, assume that the owner user's permissions have changed from <<security-api-bulk-update-api-keys-examples-user-permissions, the original permissions>> to:
+
+[source,js]
+--------------------------------------------------
+{
+  "cluster": ["manage_security"],
+  "index": [
+    {
+      "names": ["*"],
+      "privileges": ["read"]
+    }
+  ]
+}
+--------------------------------------------------
+// NOTCONSOLE
+
+The following request auto-updates the snapshot of the user's permissions associated with the two API keys.
+
+[source,console]
+----
+POST /_security/api_key/_bulk_update
+{
+  "ids": [
+    "VuaCfGcBCdbkQm-e5aOx",
+    "H3_AhoIBA9hmeQJdg7ij"
+  ]
+}
+----
+// TEST[skip:api key ids not available]
+
+Which returns the response:
+
+[source,console-result]
+----
+{
+  "updated": [
+    "VuaCfGcBCdbkQm-e5aOx",
+    "H3_AhoIBA9hmeQJdg7ij"
+  ],
+  "noops": []
+}
+----
+
+Resulting in the following effective permissions for both API keys:
+
+[source,js]
+--------------------------------------------------
+{
+  "cluster": ["manage_security"],
+  "index": [
+    {
+      "names": ["*"],
+      "privileges": ["read"]
+    }
+  ]
+}
+--------------------------------------------------
+// NOTCONSOLE
+
+If any API keys fail to update, error details are included in the `errors` field.
+For example:
+
+[[security-api-bulk-update-api-keys-examples-errors]]
+[source,js]
+--------------------------------------------------
+{
+  "updated": ["VuaCfGcBCdbkQm-e5aOx"],
+  "noops": [],
+  "errors": { <1>
+    "count": 3,
+    "details": {
+       "g_PqP4IBcBaEQdwM5-WI": { <2>
+         "type": "resource_not_found_exception",
+         "reason": "no API key owned by requesting user found for ID [g_PqP4IBcBaEQdwM5-WI]"
+       },
+       "OM4cg4IBGgpHBfLerY4B": {
+         "type": "illegal_argument_exception",
+         "reason": "cannot update invalidated API key [OM4cg4IBGgpHBfLerY4B]"
+       },
+       "Os4gg4IBGgpHBfLe2I7j": {
+         "type": "exception",
+         "reason": "error occurred while updating API key",
+         "caused_by": { <3>
+           "type": "illegal_argument_exception",
+           "reason": "security index failure"
+         }
+       }
```

Comment:
Nit: This isn't a real exception. I'll probably just leave it out since it does not really add much.

Reply:
True. I just wanted there to be a `caused_by` in the example. Would you be more comfortable keeping it if we put in a more realistic caused_by?

Reply:
I am fine with a more realistic one. I guess version conflict might be a reasonable candidate.